packages/flutter_tools/lib/src/migrate/migrate_config.dart

```diff
@@ -0,0 +1,257 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:yaml/yaml.dart';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/io.dart';
+import '../cache.dart';
+import '../flutter_project_metadata.dart';
+import '../globals.dart' as globals;
+import '../project.dart';
+
+/// Represents one .migrate_config file.
+///
+/// Each platform and the root project directory includes one .migrate_config file.
```

Comment:
to clarify, will `flutter create .` include a `. migrate_config` in the app's root directory?

Reply:
is there any issue with using the existing `.metadata` file for the root project directory, and rename .migrate_config as .metadata for the platform directories?
 
Migrate is adding a few entries to the dictionary, I'm not sure if we need a dedicated file.

Reply:
the existing .metadata file is being managed via templates, while the migrate_config is being managed by the MigrateConfig class. It is possible to merge the two of course, but we would also have to add it to each platform and handle the case where we need to modify it contrary to what the template would generate.

Reply:
And yes, flutter create will create the .migrate_config file in the root as well as each platform dir.

Reply:
> the existing .metadata file is being managed via templates, while the migrate_config is being managed by the MigrateConfig class.

That sounds good.  The `.metadata` file was created with these goals in mind:
```
# This file tracks properties of this Flutter project.
# Used by Flutter tool to assess capabilities and perform upgrades etc.
#
# This file should be version controlled and should not be manually edited.
```

It sounds like the use cases of `.migrate_config` should be covered by this file. Any thoughts?

Reply:
I agree that we shouldnt have too many metadata files. I'll see if we can merge the existing .metadata file with the .migrate_config file. Though the new version of .metadata will probably no longer be a read-only template generated file anymore.

Reply:
Awesome

Reply:
Here is a proposed extended `.metadata` file. This unifies separate config files in each platform dir to one single migrate section in the root .metadata file to remain consistent with the existing metadata tracking system. Since this file will now be owned by the tool, we can easily update the values in it as new platforms are added/edited.

```
# This file tracks properties of this Flutter project.
# Used by Flutter tool to assess capabilities and perform upgrades etc.
#
# This file should be version controlled.

version:
  revision: {{flutterRevision}}
  channel: {{flutterChannel}}

project_type: app

# Tracks metadata for the flutter migrate command
migrate:
  platforms:
    - platform: android
      createRevision: fj19vkla9vnlka9vni3n808v3nch8cd
      baseRevision: 93kf9v3njfa90vnidfjvn39nvi3vnie
    - platform: ios
      createRevision: fj19vkla9vnlka9vni3n808v3nch8cd
      baseRevision: 93kf9v3njfa90vnidfjvn39nvi3vnie

  # User provided section

  # List of Local paths (relative to this file) that should be
  # ignored by the migrate tool.
  #
  # Files that are not part of the templates will be ignored by default.
  unmanagedFiles:
    - lib/file1/etc.dart
    - android/my_file.java
```

Reply:
LGTM

Reply:
the only minor comment is to maybe use a noun for consistency, so `migrate` -> `migration`